tests/sles4sap/wizard_hana_install.pm

```diff
@@ -7,45 +7,92 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-# Summary: Install HANA with SAP Installation Wizard
+# Summary: Install HANA with SAP Installation Wizard. Verify installation with
+# sles4sap/hana_test
 # Maintainer: Ricardo Branco <rbranco@suse.de>
 
 use base 'sles4sap';
 use strict;
 use testapi;
-use utils 'turn_off_gnome_screensaver';
-use utils qw(type_string_slow zypper_call);
+use utils qw(type_string_slow zypper_call turn_off_gnome_screensaver);
+use ipmi_backend_utils 'use_ssh_serial_console';
 use version_utils 'is_sle';
 
 sub get_total_mem {
-    if (check_var('BACKEND', 'qemu')) {
-        return get_required_var('QEMURAM');
+    return get_required_var('QEMURAM') if (check_var('BACKEND', 'qemu'));
+    my $mem = script_output q@grep ^MemTotal /proc/meminfo | awk '{print $2}'@;
+    $mem /= 1024;
+    return $mem;
+}
+
+sub try_reclaiming_space {
+    my $output = script_output q@parted /dev/sda print free | awk '/Free Space/ {print $3}' | tail -1@;
+    $output =~ m/([0-9\.]+)([A-Z])B/i;
+    my $free  = $1;
+    my $units = uc($2);
+    if ($units eq 'T') {
+        $free *= 1024;
     }
-    my $mem = 0;
-    open(my $MEMINFO, "<", "/proc/meminfo") or die("Could not open /proc/meminfo");
-    while (<$MEMINFO>) {
-        $mem = int((split(/:\s+/))[1] / 1024) if (/^MemTotal:/);
-        last;
+    elsif ($units ne 'G') {
+        # Assume there's no space available if units are not T or G
+        $free = 0;
     }
-    close $MEMINFO;
-    return $mem;
+    # Only attempt to reclaim space from /dev/system/root if there's not enough free space available
+    return if ($free >= 70);
+
+    $output = script_output 'df -h --output=size,used,avail / | tail -1';
+    my ($root_size, $root_used, $root_free) = split(/\s+/, $output);
+    $root_size =~ s/([A-Z])$//i;
+    $root_free =~ s/([A-Z])$//i;
+    $units = $1;
+    if ($units eq 'T') {
+        $root_free *= 1024;
+    }
+    elsif ($units ne 'G') {
+        # Asume there's not enough free space on /dev/system/root if units are not T or G
+        $root_free = 0;
+    }
+    # Always leave at least 25Gb free on /dev/system/root for packages and Hana
+    $root_free -= 25;
+    $root_size -= $root_free if ($root_free > 0);
+    if ($root_size >= ($root_used + 25)) {
+        assert_script_run "btrfs filesystem resize $root_size$units /";
+        assert_script_run "lvreduce --yes --force --size $root_size$units /dev/system/root";
```

Comment:
In fact, I just saw that this could be done only with `lvreduce` by using `--resizefs` option.
Maybe for a next change ;-)